cw-orch-daemon/src/builder.rs

```diff
@@ -21,76 +24,30 @@ pub const DEFAULT_DEPLOYMENT: &str = "default";
 ///     .await.unwrap();
 /// # })
 /// ```
-pub struct DaemonAsyncBuilderBase<Sender: SenderTrait = Wallet> {
+pub struct DaemonAsyncBuilder {
     // # Required
-    pub(crate) chain: Option<ChainInfoOwned>,
+    pub(crate) chain: ChainInfoOwned,
     // # Optional
     pub(crate) deployment_id: Option<String>,
     pub(crate) state_path: Option<String>,
     /// State from rebuild or existing daemon
     pub(crate) state: Option<DaemonState>,
     pub(crate) write_on_change: Option<bool>,
 
-    // Sender options
-
-    // # Optional indicated sender
-    pub(crate) sender: Option<Sender>,
-
-    /// Specify Daemon Sender Options
-    pub(crate) sender_options: Sender::SenderOptions,
+    pub(crate) mnemonic: Option<String>,
 }
 
-pub type DaemonAsyncBuilder = DaemonAsyncBuilderBase<Wallet>;
-
-impl<Sender: SenderTrait> Default for DaemonAsyncBuilderBase<Sender> {
-    fn default() -> Self {
+impl DaemonAsyncBuilder {
+    pub fn new(chain: impl Into<ChainInfoOwned>) -> Self {
         Self {
-            chain: Default::default(),
-            deployment_id: Default::default(),
-            sender_options: Default::default(),
-            sender: Default::default(),
-            state_path: Default::default(),
-            state: Default::default(),
-            write_on_change: Default::default(),
+            chain: chain.into(),
+            deployment_id: None,
+            state_path: None,
+            state: None,
+            write_on_change: None,
+            mnemonic: None,
         }
     }
-}
-
-impl DaemonAsyncBuilder {
-    /// Specifies whether authz should be used with this daemon
-    pub fn authz_granter(&mut self, granter: impl ToString) -> &mut Self {
-        self.sender_options.set_authz_granter(granter);
-        self
-    }
-
-    /// Specifies whether a fee grant should be used with this daemon
-    pub fn fee_granter(&mut self, granter: impl ToString) -> &mut Self {
-        self.sender_options.set_fee_granter(granter);
-        self
-    }
-
-    /// Specifies the hd_index of the daemon sender
-    pub fn hd_index(&mut self, index: u32) -> &mut Self {
-        self.sender_options.hd_index = Some(index);
-        self
-    }
-
-    /// Set the mnemonic to use with this chain.
-    /// Defaults to env variable depending on the environment.
-    ///
-    /// Variables: LOCAL_MNEMONIC, TEST_MNEMONIC and MAIN_MNEMONIC
-    pub fn mnemonic(&mut self, mnemonic: impl ToString) -> &mut Self {
-        self.sender_options.mnemonic = Some(mnemonic.to_string());
-        self
-    }
-}
-
-impl<Sender: SenderTrait> DaemonAsyncBuilderBase<Sender> {
-    /// Set the chain the daemon will connect to
-    pub fn chain(&mut self, chain: impl Into<ChainInfoOwned>) -> &mut Self {
-        self.chain = Some(chain.into());
-        self
-    }
```

Comment:
Removing `.chain()` from the builder makes it harder to use same daemon or daemonbuilder for different chains daemons.

Maybe just adding `chain: Option<ChainInfoOwned>` to rebuild would be enough